backend/src/main/kotlin/cz/loono/backend/db/model/HealthcareProvider.kt

```diff
@@ -0,0 +1,158 @@
+    val houseNumber: String = "",
+
+    @Column(nullable = false, columnDefinition = "TEXT")
+    val region: String = "",
+
+    @Column(nullable = false, columnDefinition = "TEXT")
+    val regionCode: String = "",
+
+    @Column(nullable = false, columnDefinition = "TEXT")
+    val district: String = "",
+
+    @Column(nullable = false, columnDefinition = "TEXT")
+    val districtCode: String = "",
+
+    @Column(columnDefinition = "TEXT")
+    val administrativeDistrict: String? = null,
+
+    @Column(columnDefinition = "TEXT")
+    val phoneNumber: String? = null,
+
+    @Column(columnDefinition = "TEXT")
+    val fax: String? = null,
+
+    @Column(columnDefinition = "TEXT")
+    val email: String? = null,
+
+    @Column(columnDefinition = "TEXT")
+    val website: String? = null,
+
+    @Column(nullable = false, columnDefinition = "TEXT")
+    val ico: String = "",
+
+    @Column(nullable = false, columnDefinition = "TEXT")
+    val personTypeCode: String = "",
+
+    @Column(columnDefinition = "TEXT")
+    val lawyerFormCode: String? = null,
+
+    @Column(columnDefinition = "TEXT")
+    val layerForm: String? = null,
+
+    @Column(nullable = false, columnDefinition = "TEXT")
+    val personType: String = "",
+
+    @Column(columnDefinition = "TEXT")
+    val hqRegion: String? = null,
+
+    @Column(columnDefinition = "TEXT")
+    val hqRegionCode: String? = null,
+
+    @Column(columnDefinition = "TEXT")
+    val hqDistrict: String? = null,
+
+    @Column(columnDefinition = "TEXT")
+    val hqDistrictCode: String? = null,
+
+    @Column(columnDefinition = "TEXT")
+    val hqCity: String? = null,
+
+    @Column(columnDefinition = "TEXT")
+    val hqPostalCode: String? = null,
+
+    @Column(columnDefinition = "TEXT")
+    val hqStreet: String? = null,
+
+    @Column(columnDefinition = "TEXT")
+    val hqHouseNumber: String? = null,
+
+    @Column(nullable = false, columnDefinition = "TEXT")
+    val specialization: String? = null,
+
+    @ManyToMany
+    @JoinTable(
+        name = "healthcare_provider_category",
+        joinColumns = [
+            JoinColumn(name = "location_id", referencedColumnName = "location_id"),
+            JoinColumn(name = "institution_id", referencedColumnName = "institution_id")
+        ],
+        inverseJoinColumns = [JoinColumn(name = "category", referencedColumnName = "value")]
+    )
+    val category: Set<HealthcareCategory> = mutableSetOf(),
+
+    @Column(columnDefinition = "TEXT")
+    val careForm: String? = null,
+
+    @Column(columnDefinition = "TEXT")
+    val careType: String? = null,
+
+    @Column(columnDefinition = "TEXT")
+    val substitute: String? = null,
+
+    @Column(nullable = false, columnDefinition = "TEXT")
+    val lat: String = "",
+
+    @Column(nullable = false, columnDefinition = "TEXT")
+    val lng: String = ""
+) {
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
```

Comment:
If we are following DDD, we should compare entities by ID only.

Discuss with Martin.

Reply:
Generally, we should not compare entities only by ID. Preference is to compare them {equals, hashcode} by their [business key](https://vladmihalcea.com/hibernate-facts-equals-and-hashcode/).

Reply:
Very good, that's what I meant. I usually provide my own UUIDs for entities (= the business key) and reuse them as primary keys. I forgot that we utilize autogenerated PK in this project 🤦

Reply:
Maybe I'm odd. Firstly I had an ID column. But Hibernate and JPA couldn't have updated the records. The record is identified by the composite key of locationId and institutionId.